src/mbgl/renderer/paint_parameters.cpp

```diff
@@ -92,29 +92,37 @@ void PaintParameters::clearStencil() {
 namespace {
 
 // Detects a difference in keys of renderTiles and tileClippingMaskIDs
-bool tileIDsIdentical(const RenderTiles& renderTiles, const std::map<UnwrappedTileID, int32_t>& tileClippingMaskIDs) {
-    assert(renderTiles);
-    assert(std::is_sorted(renderTiles->begin(), renderTiles->end(), [](const RenderTile& a, const RenderTile& b) {
-        return a.id < b.id;
-    }));
-    if (renderTiles->size() != tileClippingMaskIDs.size()) {
+template <typename TIter>
+bool tileIDsIdentical(TIter beg, TIter end,
+                      std::function<const UnwrappedTileID&(typename TIter::value_type)> f,
+                      const std::map<UnwrappedTileID, int32_t>& idMap) {
+    if (static_cast<std::size_t>(std::distance(beg, end)) != idMap.size()) {
         return false;
     }
-    return std::equal(
-        renderTiles->begin(), renderTiles->end(), tileClippingMaskIDs.begin(), [](const RenderTile& a, const auto& b) {
-            return a.id == b.first;
-        });
+    assert(std::is_sorted(beg, end, [&f](const auto& a, const auto& b){ return f(a)<f(b); }));
+    return std::equal(beg, end, idMap.cbegin(), [&f](const auto& ii, const auto& pair){ return f(ii) == pair.first; });
 }
 
 } // namespace
 
+void PaintParameters::renderTileClippingMasks(const std::set<UnwrappedTileID>& tileIDs) {
+    renderTileClippingMasks(tileIDs.cbegin(), tileIDs.cend(), [](const auto& ii){ return ii; });
+}
+
 void PaintParameters::renderTileClippingMasks(const RenderTiles& renderTiles) {
-    if (!renderTiles || renderTiles->empty() || tileIDsIdentical(renderTiles, tileClippingMaskIDs)) {
+    renderTileClippingMasks((*renderTiles).cbegin(), (*renderTiles).cend(), [](const auto& ii){ return ii.get().id; });
+}
+
+template <typename TIter>
+void PaintParameters::renderTileClippingMasks(TIter beg, TIter end,
+                                              std::function<UnwrappedTileID(const typename std::iterator_traits<TIter>::value_type&)> f) {
+    if (tileIDsIdentical(beg, end, f, tileClippingMaskIDs)) {
         // The current stencil mask is for this source already; no need to draw another one.
         return;
     }
 
-    if (nextStencilID + renderTiles->size() > 256) {
+    const auto count = std::distance(beg, end);
+    if (nextStencilID + count > 256) {
```

Comment:
I suspect this should be `>255`.  The `mask` value passed to `glStencilFunc` is 255, so I don't see how a value of 256 could work, unless it's meant to be equivalent to 0.